xarray/backends/netCDF4_.py

```diff
@@ -402,7 +423,20 @@ def prepare_variable(self, name, variable, check_encoding=False,
         for k, v in iteritems(attrs):
             # set attributes one-by-one since netCDF4<1.0.10 can't handle
             # OrderedDict as the input to setncatts
-            nc4_var.setncattr(k, v)
+            if _is_list_of_strings(v):
+                # encode as NC_STRING if attr is list of strings
+                try:
+                    nc4_var.setncattr_string(k, v)
+                except AttributeError:
+                    # Inform users with old netCDF that does not support
+                    # NC_STRING that we can't serialize lists of strings
+                    # as attrs
+                    msg = ('Attributes which are lists of strings are not '
+                           'supported with this version of netCDF. Please '
+                           'upgrade to netCDF4-python 1.2.4 or greater.')
+                    raise AttributeError(msg)
+            else:
+                nc4_var.setncattr(k, v)
```

Comment:
Could you consolidate this logic into a helper function (e.g., `_set_attribute()`) rather than repeating it twice?